code/game/jobs/job_controller.dm

```diff
@@ -355,19 +355,19 @@ var/global/datum/controller/occupations/job_master
 
 		Debug("DO, Running AC2")
 
-		// For those who wanted to be civilians if their preferences were filled, here you go.
-		for(var/mob/new_player/player in unassigned)
-			if(player.client.prefs.alternate_option == BE_ASSISTANT)
-				Debug("AC2 Assistant located, Player: [player]")
-				AssignRole(player, "Civilian")
-
+		// Antags, who have to get in, come first
 		for(var/mob/new_player/player in unassigned)
 			if(player.mind.special_role)
 				GiveRandomJob(player)
+				if(player in unassigned)
+					AssignRole(player, "Civilian")
 
-		//For ones returning to lobby
+		// Then we assign what we can to everyone else.
 		for(var/mob/new_player/player in unassigned)
-			if(player.client.prefs.alternate_option == RETURN_TO_LOBBY)
+			if(player.client.prefs.alternate_option == BE_ASSISTANT)
+				Debug("AC2 Assistant located, Player: [player]")
+				AssignRole(player, "Civilian")
+			else if(player.client.prefs.alternate_option == RETURN_TO_LOBBY)
 				player.ready = 0
 				unassigned -= player

```

Comment:
Then, loop through all unassigned players again - no more spots are open. If they want to be a civilian, make them a civilian. Otherwise, set them back into the lobby